ui/org.eclipse.pde.launching/src/org/eclipse/pde/launching/JUnitEclipsePluginValidationOperation.java

```diff
@@ -0,0 +1,80 @@
+package org.eclipse.pde.launching;
+
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.Map;
+import java.util.Set;
+
+import org.eclipse.debug.core.ILaunchConfiguration;
+import org.eclipse.pde.core.plugin.IPluginModelBase;
+import org.eclipse.pde.internal.launching.launcher.EclipsePluginValidationOperation;
+import org.osgi.framework.Version;
+import org.osgi.framework.hooks.resolver.ResolverHook;
+import org.osgi.framework.wiring.BundleCapability;
+import org.osgi.framework.wiring.BundleRequirement;
+import org.osgi.framework.wiring.BundleRevision;
+
+class JUnitEclipsePluginValidationOperation extends EclipsePluginValidationOperation implements ResolverHook {
+
+	private static final Object[] EMPTY = new Object[0];
+
+	private final Map<Object, Object[]> fJunitErrors = new HashMap<>(2);
+	private final int junitVersion;
+
+	@SuppressWarnings("restriction")
+	public JUnitEclipsePluginValidationOperation(ILaunchConfiguration configuration, Set<IPluginModelBase> models) {
+		super(configuration, models);
+		setResolverHook(this);
+		org.eclipse.jdt.internal.junit.launcher.ITestKind testKind = org.eclipse.jdt.internal.junit.launcher.JUnitLaunchConfigurationConstants.getTestRunnerKind(configuration);
+		if (org.eclipse.jdt.internal.junit.launcher.TestKindRegistry.JUNIT5_TEST_KIND_ID.equals(testKind.getId())) {
+			junitVersion = 5;
+		} else {
+			junitVersion = -1;
+		}
+	}
+
+	@Override
+	public void filterMatches(BundleRequirement requirement, Collection<BundleCapability> candidates) {
+		String requirementName = requirement.getRevision().getSymbolicName();
+		if (!requirementName.startsWith("junit")) { //$NON-NLS-1$
+			Iterator<BundleCapability> iterator = candidates.iterator();
+			while (iterator.hasNext()) {
+				BundleCapability candidate = iterator.next();
+				BundleRevision candidateRevision = candidate.getRevision();
+				String name = candidateRevision.getSymbolicName();
+				Version version = candidateRevision.getVersion();
+				if (junitVersion == 5 && version.getMajor() == 6 && (name.startsWith("junit"))) { //$NON-NLS-1$
+					// TODO: externalize
+					String error = requirementName + " depends on " + name + "_" + version; //$NON-NLS-1$ //$NON-NLS-2$
+					fJunitErrors.put(error, EMPTY);
```

Comment:
Can we not better put in the requirement resource here and as a value the error message?


```suggestion
					fJunitErrors.put(requirement.getRevision(), "Depends on a different JUnit Version than the launch");
```

Reply:
That doesn't seem to include the error message:

<img width="1217" height="646" alt="Screenshot_20251106_140259" src="https://github.com/user-attachments/assets/ab94e590-b6ef-4ecb-9362-07561d573456" />